Integrations/PerceptionPoint/PerceptionPoint.py

```diff
@@ -0,0 +1,177 @@
+                        {'demisto_param': 'fetch_spam',
+                         'req_pname': VERBOSE_VERDICT_PARAM,
+                         'req_pval': SPAM}]
+
+''' HELPER FUNCTIONS '''
+
+
+def build_fetch_incident_types(fetch_select):
+    fetch_type_dict = defaultdict(list)  # type: ignore
+    for darg in FETCH_INCIDENTS_TYPE:
+        darg_input = fetch_select.get(darg['demisto_param'])
+        if darg_input:
+            fetch_type_dict[darg['req_pname']].append(darg.get('req_pval', darg_input))
+    return dict(fetch_type_dict)
+
+
+def create_incident(record):
+    record.pop('Attachment', None)
+    record['RawJSON'] = json.dumps(record)
+    return record
+
+
+def collect_incidents(**kwargs):
+    list_url = build_request_url(LIST)
+    api_res = get_pp_api_result(url=list_url, **kwargs)
+    num_of_results = api_res.get('count')
+    incidents = []  # type: list
+    api_loops = 0
+    while num_of_results and api_loops < API_MAX_LOOPS:
+        incidents += map(create_incident, api_res.get('results'))
+        if api_res.get('next'):
+            api_res = get_pp_api_result(url=api_res.get('next'))
+            num_of_results = api_res.get('count')
+        api_loops += 1
+    return incidents
+
+
+def report_incidents(incidents_list):
+    demisto.incidents(incidents_list)
+
+
+def get_pp_api_result(url, **kwargs):
+    try:
+        res = requests.get(url=url,
+                           params=kwargs,
+                           headers=HEADER,
+                           verify=SECURED)
+        res.raise_for_status()
+        try:
+            res_content = res.json()
+        except Exception:
+            res_content = {}
+        return res_content
+    except requests.exceptions.HTTPError as err:
+        if 400 <= res.status_code < 500:
+            return_error('Invalid token')
+        else:
+            return_error(err)
+    except Exception as err:
+        return_error(err)
+
+
+def build_request_url(api_action):
+    return URL.format(endpoint=API_ACTIONS_DICT.get(api_action))
+
+
+def command_fetch_incidents():
+    try:
+        args_dict = demisto.args()
+        user_params_copy = copy.deepcopy(USER_PARAMS)
+        user_params_copy.pop('insecure')
+        user_params_copy.pop('pp_token')
+        user_params_copy.pop('api_loops')
+        args_dict.update(user_params_copy)
+        req_args = build_fetch_incident_types(args_dict)
+        last_run_id = int(demisto.getLastRun().get('scan_id', 0))
+        req_args[API_CURSOR_ARG] = last_run_id
+        incidents_list = collect_incidents(**req_args)
+        report_incidents(incidents_list)
+        if incidents_list:
+            last_run_id = max(last_run_id, int(incidents_list[-1].get('Scan Id')))
+            demisto.setLastRun({'scan_id': int(last_run_id)})
+    except Exception:
+        return_error(
+            'An error occurred while trying to fetch new incidents. Please contact PerceptionPoint support')
```

Comment:
please add the str of the exception itself so you will have more info about what happened.